### DISCUSSION TITLE
Convert qmd to qmd with evaluated shortcodes

### DISCUSSION THREAD

**Discussion by royfrancis** [quarto-dev/quarto-cli#7227] [Q&A] [answered]
### Description

Is it possible to convert a qmd file like this:

````
---
title: Convert
engine: knitr

block: Block A
para: This is section A
---

## {{< meta block >}}

{{< meta para >}}

```{r}
head(iris)
```  

Some text.
````

to

````
---
title: Convert
engine: knitr
---

## Block A

This is section A

```{r}
head(iris)
```  

Some text.
````

```
▸ quarto --version
1.3.433
```

**Comment by cscheid** [comment]
Do you mean without evaluating the executable content in the r code cells? That should be possible by setting the engine to "markdown" instead of "knitr", and the file output format to `gfm+yaml_metadata_block`. It gets you pretty close (there's an extra space in front of `{r}`):

````
$ quarto render 7227.qmd --to gfm+yaml_metadata_block -o -

pandoc -o /var/folders/nm/m64n9_z9307305n0xtzpp54m0000gn/T/quarto-session480b7bcb/3f8e069d/a64214f6.md
  to: >-
    commonmark+autolink_bare_uris+emoji+footnotes+gfm_auto_identifiers+pipe_tables+strikeout+task_lists+tex_math_dollars+yaml_metadata_block
  standalone: true
  default-image-extension: png

metadata
  title: Convert
  block: Block A
  para: This is section A

---
title: Convert
engine: markdown

block: Block A
para: This is section A
---

## Block A

This is section A

``` {r}
head(iris)
```
````

**Comment by mcanouil** [comment, marked as answer]
The following should do what you want from your document without editing it:

```bash
quarto render index.qmd --to markdown --output index2.qmd --metadata engine:markdown
```

Edit: as pointing out `markdown` adds explicit anchor headings whilst `gfm` does not, thus you might prefer:

```bash
quarto render index.qmd --to gfm+yaml_metadata_block --output index2.qmd --metadata engine:markdown
```

EDIT2: `markdown` uses `yaml_metadata_block` by default, while `gfm` does not.

**Reply by cscheid** [reply]
Just as a minor point of clarification, the difference between `markdown` and `gfm` is in some of the other defaults we set. For example, `markdown` will add identifiers to headings, while `gfm` won't. There might be some other things that behave better in `markdown` than `gfm`, so make sure to check.

**Reply by mcanouil** [reply]
Indeed, It reminds me of a discussion about `md` (which is a Quarto internal format not synonymous to `markdown`).

**Reply by mcanouil** [reply]
To remove anchor attributes:

```bash
quarto render index.qmd --to markdown-header_attributes --output index2.qmd --metadata engine:markdown
```

See <https://pandoc.org/MANUAL.html> for all extensions.
